typings/index.d.ts

```diff
@@ -1014,6 +1033,29 @@ export class Intents extends BitField<IntentsString> {
   public static resolve(bit?: BitFieldResolvable<IntentsString, number>): number;
 }
 
+export type GuildCacheState = 'cached' | 'raw' | 'present';
+
+type CacheTypeReducer<
+  State extends GuildCacheState,
+  CachedType,
+  RawType = CachedType,
+  PresentType = CachedType | RawType,
+  Fallback = PresentType | null,
+> = State extends 'cached'
+  ? CachedType
+  : State extends 'raw'
+  ? RawType
+  : State extends 'present'
+  ? PresentType
+  : Fallback;
+
+export interface GuildInteraction<Cached extends GuildCacheState = GuildCacheState> extends Interaction {
+  guildId: CacheTypeReducer<Cached, Snowflake>;
```

Comment:
Is there any reason you don't just mark this as a Snowflake anyways? If it's a *Guild*Interaction, guildId is always present